server/src/main/java/org/elasticsearch/cluster/LocalNodeMasterListener.java

```diff
@@ -33,5 +33,14 @@
      * Called when the local node used to be the master, a new master was elected and it's no longer the local node.
      */
     void offMaster();
+
+    @Override
+    default void clusterChanged(ClusterChangedEvent event) {
+        if (!event.previousState().nodes().isLocalNodeElectedMaster() && event.localNodeMaster()) {
```

Comment:
NIT: Maybe move this to our standard style of `== false` and also cache `final boolean wasMaster = event.previousState().nodes().isLocalNodeElectedMaster()` to make it a little shorter?